core/arch/arm/plat-stm32mp1/main.c

```diff
@@ -277,20 +277,58 @@ early_init_late(set_all_gpios_non_secure);
 #endif /* CFG_STM32_GPIO */
 #endif /* CFG_STM32MP13 */
 
-#if defined(CFG_STM32MP15) && defined(CFG_TZSRAM_START)
+#ifdef CFG_STM32MP15
 /*
- * TZSRAM address range shall comply with one of the below cases:
- * - TZSRAM fully fits inside SYSRAM secure address range
- * - TZSRAM fully fits inside the full SYSRAM and spread inside SRAMs
- * - TZSRAM fully fits inside SRAMs address range
- * - TZSRAM fully fits in DDR (above SRAMs) for debug and test purpose
+ * This concerns OP-TEE pager for STM32MP1 to use secure internal
+ * RAMs to execute. TZSRAM refers the TZSRAM_BASE/TZSRAM_SIZE
+ * used in boot.c to locate secure unpaged memory.
+ *
+ * STM32MP15 variants embed 640kB of contiguous securable SRAMs
+ *
+ *  +--------------+ <-- SYSRAM_BASE
+ *  |              |     lower part can be assigned to secure world
+ *  | SYSRAM 256kB |     4kB granule boundary
+ *  |              |     upper part can be assigned to secure world
+ *  +--------------+ <-- SRAM1_BASE (= SYSRAM_BASE + SYSRAM_SIZE)
+    |              |
+ *  | SRAM1  128kB |     full range assigned to non-secure, secure or Cortex-M4
+ *  +--------------+ <-- SRAM2_BASE (= SRAM1_BASE + SRAM1_SIZE)
+    |              |
+ *  | SRAM2  128kB |     full range assigned to non-secure, secure or Cortex-M4
+ *  +--------------+ <-- SRAM3_BASE (= SRAM2_BASE + SRAM2_SIZE)
+ *  | SRAM3   64kB |     full range assigned to non-secure, secure or Cortex-M4
+ *  +--------------+ <-- SRAM4_BASE (= SRAM3_BASE + SRAM3_SIZE)
+ *  | SRAM4   64kB |     full range assigned to non-secure, secure or Cortex-M4
+ *  +--------------+ <-- SRAM4_BASE + SRAM4_SIZE
+ *
+ * If SRAMx memories are not used for the companion Cortex-M4
+ * processor, OP-TEE can use this memory.
+ *
+ * SYSRAM configuration for secure/non-secure boundaries require the
+ * secure SYSRAM memory to start the SYSRAM physical address and grow
+ * from there and the non-secure SYSRAM to lie in SYSRAM last 4KB pages.
+ *
+ * SRAM1, SRAM2, SRAM3 and SRAM4 are independently assigned to secure
+ * or non-secure world, or possibly to Cortex-M only.
+ *
+ * Using non-secure SYSRAM or one of the SRAMx for SCMI message communication
+ * can be done using CFG_STM32MP1_SCMI_SHM_BASE/CFG_STM32MP1_SCMI_SHM_SIZE.
+ * This imposes related memory area is assigned to non-secure world.
+
+ * Using secure internal memories (SYSRAM and/or some SRAMx) with STM32MP15
+ * shall meet this constraints.
  */
```

Comment:
A paragraph on the necessary continuity of the secure memory across these RAMs would be nice.
It may be unclear the entire SRAM is configured as non-secure memory when SCMI SHM is part of it.